server_addon/maya/client/ayon_maya/plugins/publish/extract_maya_scene_raw.py

```diff
@@ -88,17 +88,30 @@ def process(self, instance):
         )
         with maintained_selection():
             cmds.select(selection, noExpand=True)
-            cmds.file(path,
-                      force=True,
-                      typ="mayaAscii" if self.scene_type == "ma" else "mayaBinary",  # noqa: E501
-                      exportSelected=True,
-                      preserveReferences=attribute_values[
-                          "preserve_references"
-                      ],
-                      constructionHistory=True,
-                      shader=True,
-                      constraints=True,
-                      expressions=True)
+            if instance.data.get("shader", True):
+                with shader(selection, shadingEngine="initialShadingGroup"):
+                    cmds.file(path,
+                              force=True,
+                              typ="mayaAscii" if self.scene_type == "ma" else "mayaBinary",  # noqa: E501
+                              exportSelected=True,
+                              preserveReferences=attribute_values[
+                                  "preserve_references"
+                                  ],
+                              constructionHistory=True,
+                              shader=instance.data.get("shader", True),
+                              expressions=True)
+            else:
+                cmds.file(path,
+                          force=True,
+                          typ="mayaAscii" if self.scene_type == "ma" else "mayaBinary",  # noqa: E501
+                          exportSelected=True,
+                          preserveReferences=attribute_values[
+                              "preserve_references"
+                              ],
+                          constructionHistory=True,
+                          shader=True,
+                          constraints=True,
+                          expressions=True)
```

Comment:
Preferably we don't need to duplicate all these lines of code.

How about using `ExitStack`?

```python
with contextlib.ExitStack() as stack:
    if not instance.data.get("shader", True):
        # Fix bug where export without shader may import the geometry 'green' 
        # due to the lack of any shader on import.
        stack.enter_context(shader(selection, shadingEngine="initialShadingGroup"))

    cmds.file(path,
              force=True,
              typ="mayaAscii" if self.scene_type == "ma" else "mayaBinary",
              exportSelected=True,
              preserveReferences=attribute_values["preserve_references"],
              constructionHistory=True,
              shader=instance.data.get("shader", True),
              constraints=True,
              expressions=True)
```